src/views/signup.js

```diff
@@ -1,179 +1,179 @@
-import React from 'react';
-import { Form, Field, Formik, ErrorMessage } from 'formik';
-import dwellinglyLogo from '../assets/images/dwellingly_logo.png';
-import UserContext from '../UserContext';
-import { Redirect } from 'react-router';
-//import Header from '../components/Header/index';
-import axios from 'axios';
+/* eslint-disable react/jsx-filename-extension */
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  Form, Field, Formik, ErrorMessage,
+} from "formik";
+import { Redirect } from "react-router";
+import axios from "axios";
+import * as Yup from "yup";
+import UserContext from "../UserContext";
+import dwellinglyLogo from "../assets/images/dwellingly_logo.png";
 
-function isValidEmail(email) {
-  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-  return re.test(String(email).toLowerCase());
-}
-
-export class SignupForm extends React.Component {
-
-  async signup(firstName, lastName, email, phone, password, confirmPassword){
-
-    if (password !== confirmPassword){
-      alert("Passwords Don't Match");
-      return;
-    }
-
-    return axios.post("/api/register", {
-      firstName: firstName,
-      lastName: lastName,
-      email: email,
-      phone: phone,
-      password: password,
-      confirmPassword: confirmPassword
-    })
-    .then((response) => {
-      if (response){
-        const success = 'Account Created Successfully!'
-        console.log(success);
-        alert(success);
-        //Redirect to login using the react router history
-        this.props.history.push('/login');
-        return response;
-      }
-    })
-    .catch((error) => {
-      if (error.response){
-        console.log(error.response.data.message)
-        alert(error.response.data.message)
+const SignupForm = ({ history }) => {
+  const signup = async (firstName, lastName, email, phone, password, confirmPassword) => {
+    let response;
+    try {
```

Comment:
Nice! I think try/catch blocks are something that's needed in more of our components.

Reply:
For async operations, it can be a more readable alternative to Promise chain handling.